doc/user_guide/features/formatting_code/index.rst

```diff
@@ -0,0 +1,126 @@
+.. _formatting_code:
+
+Formatting code
+===============
+
+.. toctree::
+    :maxdepth: 2
+
+    troubleshooting
+
+The PTB automatically formats code and ensures via a step in the ``checks.yml`` GitHub
+workflow that committed code adheres to these standards. The goals of this are to
+improve the readability and maintainability of the code and to provide a uniform
+experience across projects.
+
+.. _formatting_sessions:
+
+Nox sessions
+++++++++++++
+
+For autoformatting, the following tools are used:
+
+* `black <https://black.readthedocs.io/en/stable/the_black_code_style/index.html>`__ -
+  formats Python code to maintain consistent styling standards.
+* `isort <https://pycqa.github.io/isort/index.html>`__ - organizes and formats Python
+  import statements alphabetically and by type (from __future__, standard library
+  packages, third party packages, and local application imports).
+* `pyupgrade <https://pypi.org/project/pyupgrade/>`__ - upgrades syntax for newer
+  versions of the Python language.
+
+In the PTB, these tools are bundled into nox sessions to ensure that they are run in a
+deterministic manner.
+
++--------------------+------------------+------------------------------------+
+| Nox session        | CI Usage         | Action                             |
++====================+==================+====================================+
+| ``project:fix``    | No               | Applies code formatting changes    |
++--------------------+------------------+------------------------------------+
+| ``project:format`` | ``checks.yml``   | Checks that current code does not  |
+|                    |                  | need to be re-formatted            |
++--------------------+------------------+------------------------------------+
+
+.. _pre_commit:
+
+pre-commit
```

Comment:
Hm, the pre-commit does not only run formatting checks, but also linter and type checks.
Wouldn't it be then better to move this section into some other (more general) documentation file?